### ISSUE TITLE
CFAST freezes with this input file

### ISSUE THREAD

**Issue by akrasuski** [firemodels/cfast#805]
Could you guide me please what is wrong with this input file that cfast is unable to calculate? It freezes in 10 s. 
[cfast.txt](https://github.com/firemodels/cfast/files/1341355/cfast.txt)

**Comment by andytamhk168** [comment]
Let me take a look first.

**Comment by andytamhk168** [comment]
@akrasuski: I have looked into your input file. What I did is that I break down your problem/simulation into smaller problems so that I can look into each and everyone of the features/components you have used in your case. 

It happens to me that some of your vents are causing the "freeze" problem. Specifically, CFAST is working extra hard when the HVENT (HVENT,22,49,1,0.85,2,0,0.89,,3,TIME,,,0,1,0,1) is being included in the simulation. I notice that CFAST is reducing the calculation time-step significantly for convergence.

Since you are dealing with a simulation with relatively complex geometry, I would suggest that you start your simulation with a simpler case. For example, you can first configure your case with only a number of compartments surrounded by the compartment where the fire in located at. Then, you can observe and analyze the results. And then you can decide how you would like to expand your geometry. 

Please give this approach a try first.

Also, I notice that you have exceedingly number of inputs for many of your vents. Please correct that.

Lastly, please find attached the .in file. It contains a troubleshooting case where a significant reduction of time-step is required in CFAST. For this particular case, it took about 500 seconds to complete the simulation.

[troubleshooting3.txt](https://github.com/firemodels/cfast/files/1345104/troubleshooting3.txt)

**Comment by akrasuski** [comment]
@andytamhk168 Thank you very much for your help and effort analyzing this quite complex file. 

I'm sorry, I haven't added some comment in order to make my problem more clear. 

The input file comes from Monte Carlo process. So it is generated automatically. I try to model all the uncertain parameters in CFAST as probability distributions. So I draw geometry ones (as invariant), and then get the parameters for specific simulation from the distributions.  These random variables are related to the fire location, fire parameters, vents open/close position and many others. 
Therefore expert judgment based simplification and expansion you mention is difficult to implement in my workflow. 

I hope that having more insight to my problem we will also find some solution.

**Comment by andytamhk168** [comment]
I believe one of the solutions that would help you to get around for the convergence matter will be clicking "Update" in the graphical user interface while you are running the case. The simulation moves on the next time-step. However, I do not suggest that because the accuracy of the results is highly uncertain since the solutions have not been converged, . 

Even though the different compartments and vents are generated by MC, I suggest that you will still have to look into each of the compartments and vents and make sure that they make sense. Otherwise, you may have to wait for a long time and what you get from the simulation may not give you any constructive insights. 

In short, I do not see there is a bug in CFAST causing the long computational time.

**Comment by andytamhk168** [comment]
I believe one of the solutions that would help you to get around for the convergence matter will be clicking "Update" in the graphical user interface while you are running the case. The simulation moves on the next time-step. However, I do not suggest that because the accuracy of the results is highly uncertain since the solutions have not been converged, .

Even though the different compartments and vents are generated by MC, I suggest that you will still have to look into each of the compartments and vents and make sure that they make sense. Otherwise, you may have to wait for a long time since you are asking CFAST to try hard to simulate extremely small heat and mass transfer process in those compartments. And, what you get from the simulation may not give you any constructive insights.

In short, I do not see there is a bug in CFAST causing the long computational time.

**Comment by rpeacoc** [comment]
As you know, there are (hopefully) rare cases where the ODE solver has a particularly difficult time solving the equations and cuts the time step very small.  The case isn't freezing, just taking very small time steps.  

When running cases from a Monte Carlo simulation, we usually give the run a maximum time to run.  If it doesn't run in that time, we drop that case and do another sample.  This may also work for you if it fits within your sampling scheme.
